relayer/relays/parachain/beefy-listener.go

```diff
@@ -82,27 +86,24 @@ func (li *BeefyListener) Start(ctx context.Context, eg *errgroup.Group) error {
 	eg.Go(func() error {
 		defer close(li.tasks)
 
-		// Subscribe NewMMRRoot event logs and fetch parachain message commitments
-		// since latest beefy block
-		beefyBlockNumber, _, err := li.fetchLatestBeefyBlock(ctx)
-		if err != nil {
-			return fmt.Errorf("fetch latest beefy block: %w", err)
-		}
-
-		err = li.doScan(ctx, beefyBlockNumber)
-		if err != nil {
-			return fmt.Errorf("scan for sync tasks bounded by BEEFY block %v: %w", beefyBlockNumber, err)
-		}
+		ticker := time.NewTicker(time.Second*60 + time.Duration(rand.Intn(30))*time.Second)
+		for {
+			beefyBlockNumber, _, err := li.fetchLatestBeefyBlock(ctx)
+			if err != nil {
+				return fmt.Errorf("fetch latest beefy block: %w", err)
+			}
 
-		err = li.subscribeNewMMRRoots(ctx)
-		if err != nil {
-			if errors.Is(err, context.Canceled) {
+			err = li.doScan(ctx, beefyBlockNumber)
+			if err != nil {
+				return fmt.Errorf("scan for sync tasks bounded by BEEFY block %v: %w", beefyBlockNumber, err)
+			}
+			select {
+			case <-ctx.Done():
 				return nil
+			case <-ticker.C:
+				continue
 			}
-			return err
```

Comment:
Replace `subscribeNewMMRRoots` with a random periodical scan. in case one relayer is down and other relayers won't compete for the failed message(previously assigned to that relayer) at same time.